extensions/parquet/table/src/main/java/io/deephaven/parquet/table/location/ParquetTableLocation.java

```diff
@@ -177,36 +179,51 @@ RowGroupReader[] getRowGroupReaders() {
                 return local;
             }
             initialize();
-            return rowGroupReaders = IntStream.of(rowGroupIndices)
-                    .mapToObj(idx -> parquetFileReader.getRowGroup(idx, version))
+            local = IntStream.of(rowGroupIndices)
+                    .mapToObj(idx -> parquetFileReader.getRowGroup(idx, getVersion()))
                     .sorted(Comparator.comparingInt(rgr -> rgr.getRowGroup().getOrdinal()))
                     .toArray(RowGroupReader[]::new);
+            rowGroupReaders = local;
+            return local;
         }
     }
 
     @Override
     @NotNull
     public List<SortColumn> getSortedColumns() {
         initialize();
+        if (sortingColumns == null) {
+            sortingColumns = SortColumnInfo.sortColumns(tableInfo.sortingColumns());
+        }
         return sortingColumns;
     }
 
     @NotNull
-    Map<String, String[]> getParquetColumnNameToPath() {
+    private Map<String, GroupingColumnInfo> getGroupingColumns() {
         initialize();
-        return parquetColumnNameToPath;
+        if (groupingColumns == null) {
+            groupingColumns = Collections.unmodifiableMap(tableInfo.groupingColumnMap());
+        }
+        return groupingColumns;
     }
 
-    @Override
-    public final RowSet getRowSet() {
+    private String getVersion() {
+        initialize();
+        if (version == null) {
+            version = tableInfo.version();
+        }
+        return version;
+    }
```

Comment:
I don't think we need this as a member variable, nor as a method. We should just be able to call `tableInfo.version()` in `getRowGroupReaders` I think.